sdk/monitor/azquery/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,177 @@
+
+```go
+import azlog "github.com/Azure/azure-sdk-for-go/sdk/azcore/log"
+
+// Print log events to stdout
+azlog.SetListener(func(cls azlog.Event, msg string) {
+	fmt.Println(msg)
+})
+
+// Includes only requests and responses in credential logs
+azlog.SetEvents(azlog.EventRequest, azlog.EventResponse)
+```
+
+### Authentication errors
+
+Azure Monitor Query supports Azure Active Directory authentication. Both LogsClient and
+MetricsClient take in a `credential` as a parameter in their constructors. To provide a valid credential, you can use
+`azidentity` package. For more details on getting started, refer to
+the [README][readme_authentication]
+of Azure Monitor Query library. For details on the credential types supported in `azidentity`, see the [Azure Identity library's documentation][azidentity_docs].
+
+For more help with troubleshooting authentication errors, see the Azure Identity client library [troubleshooting guide][azidentity_troubleshooting].
+
+## Troubleshooting Logs Query
+
+### Troubleshooting authorization errors for logs query
+
+If you get an HTTP error with status code 403 (Forbidden), it means that the provided credentials does not have
+sufficient permissions to query the workspace.
+```text
+com.azure.core.exception.HttpResponseException: Status code 403, "{"error":{"message":"The provided credentials have insufficient access to perform the requested operation","code":"InsufficientAccessError","correlationId":""}}"
+```
+
+1. Check that the application or user that is making the request has sufficient permissions:
+    * You can refer to this document to [manage access to workspaces][workspace_access]
+2. If the user or application is granted sufficient privileges to query the workspace, make sure you are
+   authenticating as that user/application. If you are authenticating using the
+   [DefaultAzureCredential][default_azure_cred]
+   then check the logs to verify that the credential used is the one you expected. To enable logging, see [enable
+   client logging](#logging) section above.
+
+For more help with troubleshooting authentication errors, see the Azure Identity client library [troubleshooting guide][azidentity_troubleshooting].
+
+### Troubleshooting invalid Kusto query
+
+If you get an HTTP error with status code 400 (Bad Request), you may have an error in your Kusto query and you'll
+see an error message similar to the one below.
+
+```text
+com.azure.core.exception.HttpResponseException: Status code 400, "{"error":{"message":"The request had some invalid properties","code":"BadArgumentError","correlationId":"ff3e2a7e-e95c-4437-82cf-9b15761d0850","innererror":{"code":"SyntaxError","message":"A recognition error occurred in the query.","innererror":{"code":"SYN0002","message":"Query could not be parsed at 'joi' on line [2,244]","line":2,"pos":244,"token":"joi"}}}}"
```

Comment:
Strip out value for `correlationId`

Reply:
fixed in commit [fee0a94](https://github.com/Azure/azure-sdk-for-go/pull/19399/commits/fee0a94bfa771c52e58fed930ff0c0c2e717f004)